src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props

```diff
@@ -53,7 +53,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PackageReference Update="Microsoft.NETCore.App"
                       Condition="('$(_TargetFrameworkVersionWithoutV)' != '') And ('$(_TargetFrameworkVersionWithoutV)' >= '2.0') And ('$(PackageType)' != 'DotnetCliTool')"
                       PrivateAssets="All"
-                      Publish="true" />    
+                      Publish="true" />
+
+    <!-- Ensure the latest apphost is used when building or publishing with the apphost
+         for framework dependent applications (.NET Core 2.2 or higher). -->
+    <PackageReference Condition="'$(SelfContained)' != 'true' and '$(UseAppHost)' == 'true' and '$(HasRuntimeOutput)' == 'true'"
```

Comment:
@nguerrera Here's the aforementioned implicit reference.  I'm fine with removing this in favor of instructing users to do the following if they need the latest apphost patch:

```xml
<PackageReference Include="Microsoft.NETCore.DotNetAppHost" Version="$(LatestNetCorePatchVersion)" />
```

Reply:
cc @eerhardt @richlander @steveharter 

Main question here is: Can we depend on the 1:1 version relationship of Microsoft.NETCore.App:Microsfot.NETCore.DotNetAppHost?

Reply:
I think I'd prefer removing this as it reduces the coupling from SDK to framework packaging. My assumption is that the amount of code in the apphost is small enough that you can manually upgrade it if you actually encounter an issue.

But there's also a thread where we're discussing whether this should just work if I target .NET Core 2.1 and if we want that, then we need to do something to get a host that is not 2.1.0 by default.

And I suppose there's some risk of a security problem in the apphost itself, in which case we'd have to advise folks to edit their projects and not just rebuild with a new SDK.

Reply:
How is the reference to DotNetAppHost different between self-contained and framework-dependent apphost w.r.t. using the latest version? Why wouldn't we want the same semantics?

Reply:
Self-contained applications move the NetCore.App reference itself to latest patch level, so they already have the same version as this implicit package reference via a transitive dependency.

Reply:
@eerhardt @richlander @steveharter do we have an answer for whether or not the NetCore.DotNetAppHost package will always share a version number with NetCore.App package?

This change is relying on that, and thus far that's been true with the framework releases.  Otherwise, if we still want the implicit reference, we'll have to manage a separate version number just for the apphost.

Reply:
> do we have an answer for whether or not the NetCore.DotNetAppHost package will always share a version number with NetCore.App package?

ptal @weshaggard @leecow

It looks like the current scripts assume the same and map to "ProductVersion"

Microsoft.NETCore.DotNetAppHost: https://github.com/dotnet/core-setup/blob/c54c12905b41cd5257794a2e06f7c9b5a1dc5fe4/src/pkg/projects/Microsoft.NETCore.DotNetHostResolver/Microsoft.NETCore.DotNetHostResolver.pkgproj#L11
```
<Version>$(AppHostVersion)</Version>
```

https://github.com/dotnet/core-setup/blob/6a89975a97472a52725b343bc17bb6bef7845d16/dir.props#L95
```
<UseShippedAppHostPackage>false</UseShippedAppHostPackage>
<AppHostVersion Condition="'$(UseShippedAppHostPackage)' != 'true'">$(ProductVersion)
```

Microsoft.NETCore.App:https://github.com/dotnet/core-setup/blob/002fc811fbe09c23f525fb66b7f7156b164f2acb/src/pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj#L8
```
<Version>$(SharedFrameworkNugetVersion)</Version>
```

https://github.com/dotnet/core-setup/blob/6a89975a97472a52725b343bc17bb6bef7845d16/dir.props#L78
```
<SharedFrameworkNugetVersion>$(ProductVersion)</SharedFrameworkNugetVersion>
```

Reply:
Oh, I didn't catch that part of the question.

No, you shouldn't rely on the DotNetAppHost having the same version number as `Microsoft.NETCore.App`.  We service the runtime (`Microsoft.NETCore.App`) more often than we need to service the host.

When we serviced `1.0` and `1.1`, we serviced these packages separately. But looking at the history of `2.0` and `2.1`, it appears we've serviced the host every time the runtime was serviced. I'm not sure why that is the case, maybe @steveharter or @weshaggard would know.

But in general, if we need to fix a bug in the JIT, or in the BCL, there is no reason to ship a new app host. So in the core-setup repo, we have the ability to ship new `Microsoft.NETCore.App` versions without shipping new Host FXR, DotnetHost, or DotnetAppHost packages.

Also cc'ing @jeffschwMSFT and @vitek-karas, in case they have any "future" looking insight into the app host.

Reply:
Given this, let's just not do the implicit and for the rare case where there's a servicing fix needed for the apphost, we can have users manual ref a later version or just target the whole app to latest patch.

Reply:
I'm fine with that provided we're ok with a 2.1 targeted project publishing with a 2.2 SDK doesn't get the apphost.

Reply:
Good point. I'm personally fine with it. User could still force useapphost to true and add a ref to a newer apphost if they really need it on 2.1, right?

Reply:
Per email thread, I will now back out the implicit reference.